azavea/echo-locator
# Neighborhoods API

## Overview

Pulls neighborhood data from the Django API rather than including it in the application bundle.


### Checklist


### Demo

N/A -- Everything should look the same.


### Notes

- I have had occasional issues with the neighborhoods showing up with strange corrupted polygons, e.g.:
<img width="1075" alt="Screen Shot 2022-06-08 at 10 47 05" src="https://user-images.githubusercontent.com/447977/172646662-c7144477-75cc-4a30-80b4-91bb0e59c2d2.png">
I suspect the issue is some kind of race condition, so I'm hoping it's limited to my particular environment, but please be on the lookout for it and let me know if you encounter it.


## Testing Instructions

 * Before checking out this branch, run `./scripts/server` on `develop` and log in to the app. Take note of the neighborhood recommendations you are provided, including their routing instructions, school quality, and max rents.
 * Check out this branch and run `./scripts/update` followed by `./scripts/server`
 * Log in and log back out and confirm that you see network requests to `/api/neighborhoods/` and `/api/neighborhood-bounds/`. You may see multiple requests -- this is something we'd ideally eliminate because it indicates there are superfluous re-renders occurring in the app, but we don't have time to address that at the moment. I'll open a tech-debt issue for it.
 * Confirm that the neighborhood recommendations and attributes are the same as the ones on `develop`, except that neighborhood images will not work (this is expected and will be addressed in #494 ).

Resolves #493
